datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java

```diff
@@ -65,7 +65,7 @@
      *
      * @since 2.13
      */
-    private static final Pattern ISO8601_COLONLESS_OFFSET_REGEX = Pattern.compile("[+-][0-9]{4}");
+    private static final Pattern ISO8601_COLONLESS_OFFSET_REGEX = Pattern.compile("(?<!^)[+-][0-9]{4}");
```

Comment:
Ok, I did think of one more thing...for the regular exp., why not restrict this to the last 5 characters of the string? Two reasons this might be nice (1) limits the possibility that there is some odd combination such that there are 4 digits in the middle of the string, that's something else besides the offset, and we don't want a colon inserted there (2) performance gain - admittedly very small, but regexs can easily get out of hand. This one here is perfectly fine, but for some interesting reading on the subject: https://regular-expressions.mobi/catastrophic.html?wlr=1